src/Features/CSharp/Portable/AddParameter/CSharpAddParameterCodeFixProvider.cs

```diff
@@ -39,5 +42,40 @@ protected override ImmutableArray<string> TooManyArgumentsDiagnosticIds
 
         protected override ImmutableArray<string> CannotConvertDiagnosticIds
             => GenerateConstructorDiagnosticIds.CannotConvertDiagnosticIds;
+
+        protected override RegisterFixData<ArgumentSyntax> GetDataForFix_LanguageSpecificExpression(
+            SemanticModel semanticModel,
+            SyntaxNode node,
+            CancellationToken cancellationToken)
+        {
+            if (node is ConstructorInitializerSyntax constructorInitializer)
+            {
+                var constructorDeclaration = constructorInitializer.Parent;
+                if (semanticModel.GetDeclaredSymbol(constructorDeclaration, cancellationToken) is IMethodSymbol constructorSymbol)
+                {
+                    var type = constructorSymbol.ContainingType;
+                    if (constructorInitializer.IsKind(SyntaxKind.BaseConstructorInitializer))
+                    {
+                        // Search for fixable constructors in the base class.
+                        type = type?.BaseType;
+                    }
+
+                    if (type != null && type.IsFromSource())
+                    {
+                        var methodCandidates = type.InstanceConstructors;
+                        if (constructorInitializer.IsKind(SyntaxKind.ThisConstructorInitializer))
```

Comment:
technically, this check isn't strictly necessary.  you could just remove.

Reply:
I kept it because it makes clear that the removal is for the `this` part only. This should be easier to understand because you don't need to figure out that the removal is a no-op in the `base` case.

Reply:
👍